modules/controlfreec/1.0/controlfreec.smk

```diff
@@ -0,0 +1,157 @@
+    conda:
+        CFG["conda_envs"]["sambamba"]
+    params:
+        config = CFG["options"]["configFile"],
+        outdir = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/",
+        window = CFG["options"]["window"],
+        ploidy = CFG["options"]["ploidy"],
+        chrLen = CFG["options"]["chrLenFile"],
+        chrFiles = CFG["options"]["chrFiles"],
+        reference = CFG["options"]["gemMappabilityFile"]
+    shell:
+        "samtoolspath=$(which samtools ) ; "
+        "samtoolsPathName=$(echo $samtoolspath) ; "
+        "sambambapath=$(which sambamba ) ; "
+        "sambambaPathName=$(echo $sambambapath) ; "
+        "bedtoolspath=$(which bedtools ) ; "
+        "bedtoolsPathName=$(echo $bedtoolspath) ; "
+        "sed \"s|BAMFILE|{input.bam}|g\" {params.config} | "
+        "sed \"s|OUTDIR|{params.outdir}|g\" | "
+        "sed \"s|windowSize|{params.window}|g\" | "
+        "sed \"s|ploidyInput|{params.ploidy}|g\" | "
+        "sed \"s|chrLenFileInput|{params.chrLen}|g\" | "
+        "sed \"s|chrFilesPath|{params.chrFiles}|g\" | "
+        "sed \"s|sambambaPath|$sambambaPathName|g\" | "
+        "sed \"s|bedtoolsPath|$bedtoolsPathName|g\" | "
+        "sed \"s|samtoolsPath|$samtoolsPathName|g\" | "
+        "sed \"s|referenceFile|{params.reference}|g\" > {output}"
+
+rule _controlfreec_run:
+    input:
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/config_WGS.txt"
+    output:
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_info.txt",
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_ratio.txt",
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_CNVs"
+    conda: CFG["conda_envs"]["controlfreec"]
+    threads: CFG["threads"]["controlfreec_run"]
+    resources: mem_mb = CFG["mem_mb"]["controlfreec_run"]
+    log:
+        stdout = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/run.stdout.log",
+        stderr = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/run.stderr.log"
+    shell:
+        "freec -conf {input} > {log.stdout} 2> {log.stderr} "
+
+rule _controlfreec_calc_sig:
+    input:
+        CNVs = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_CNVs",
+        ratios = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_ratio.txt",
+    output:
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_CNVs.p.value.txt"
+    params:
+        calc_sig = CFG["software"]["FREEC_sig"]
+    threads: CFG["threads"]["calc_sig"]
+    resources: mem_mb = CFG["mem_mb"]["calc_sig"]
+    conda: CFG["conda_envs"]["controlfreec"]
+    log:         
+        stdout = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/calc_sig.stdout.log",
+        stderr = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/calc_sig.stderr.log"
+    shell:
```

Comment:
If this is an R environment should it be separate from the controlfreec executable's environment?

Reply:
I haven't ran into any issues with it so far. Controlfreec automatically includes the R environment when being installed